appstore-metadata-service/src/main/java/com/lgi/appstore/metadata/util/ApplicationPreferredHelper.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * If not stated otherwise in this file or this component's LICENSE file the
+ * following copyright and licenses apply:
+ *
+ * Copyright 2023 Liberty Global Technology Services BV
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.lgi.appstore.metadata.util;
+
+import org.jooq.JSONB;
+import org.jooq.Record10;
+import org.jooq.Record13;
+import org.jooq.Record20;
+import org.jooq.Record21;
+import org.jooq.Result;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import static com.lgi.appstore.metadata.jooq.model.tables.Application.APPLICATION;
+
+public class ApplicationPreferredHelper {
+
+    private ApplicationPreferredHelper() {
+    }
+
+    public static List<Record10<String, String, String, String, String, String, Integer, String, JSONB, Boolean>>
+    collectPreferredVersionFirstListStb(Result<Record10<String, String, String, String, String, String, Integer, String, JSONB, Boolean>> result) {
+        return result.stream()
+                .filter(record1 -> result.stream()
+                        .noneMatch(record2 -> record1 != record2
+                                && record1.get(APPLICATION.ID_RDOMAIN).equals(record2.get(APPLICATION.ID_RDOMAIN))
+                                && record2.get(APPLICATION.PREFERRED).equals(true))
+                ).collect(Collectors.toList());
+    }
+
+    public static List<Record13<String, String, String, String, String, Boolean, Boolean, Boolean, String, String, Integer, String, JSONB>>
+    collectPreferredVersionFirstForListMaintainer(Result<Record13<String, String, String, String, String, Boolean, Boolean, Boolean, String, String, Integer, String, JSONB>> result) {
+        return result.stream()
+                .filter(record1 -> result.stream()
+                        .noneMatch(record2 -> record1 != record2
+                                && record1.get(APPLICATION.ID_RDOMAIN).equals(record2.get(APPLICATION.ID_RDOMAIN))
+                                && record2.get(APPLICATION.PREFERRED).equals(true))
+                ).collect(Collectors.toList());
```

Comment:
seems collectPreferredVersionFirstListStb and collectPreferredVersionFirstForListMaintainer share the same logic, let's put it to single method and re-use

Reply:
@den314 I tried it, but there is a problem with strong typing Records type. The method needs exactly RecordX where X is a number.